File: xmp/AsyncSemaphore.js
#!/usr/bin/env node
if ( require.main === module )
{
  var gepard = require ( "gepard" ) ;

  if ( gepard.getProperty ( "help" ) )
  {
    console.log (
      "Gepard example: Semaphore, acquire a given semaphore.\n"
    + "Usage: node Semaphore [options]\n"
    + "  Options are: -Dname=<semaphore-name>, default <semaphore-name>=user:4711\n"
    ) ;
    process.exit() ;
  }
  new gepard.Admin().isRunning ( function admin_is_running ( state )
  {
    if ( ! state )
    {
      console.log ( "Not running on " + this.getHostPort() ) ;
      process.exit ( 1 ) ;
    }
    execute() ;
  });
  function execute()
  {
    var key = gepard.getProperty ( "name", "user:4711" ) ;
    var sem = new gepard.Semaphore ( key ) ;
    console.log ( "Aquiring semaphor=" + key ) ;
    sem.acquire ( function ( err )
    {
      console.log ( this.toString() ) ;
      console.log ( "Is owner: " + this.isOwner() ) ;
      console.log ( "sleep for 10 seconds" ) ;
      setTimeout ( function sleep ()
      {
        sem.release() ;
        console.log ( "Semaphore released." ) ;
      }, 10000 ) ;
    } ) ;
  }
}

File: python/AsyncSemaphore.py
#!/usr/bin/env python

from Gepard import Client, Semaphore
import time

# ==========================================================================

client = Client.getInstance()

def on_error ( err, info ):
	print ( err )

client.onError ( on_error )

def on_acquired(sem):
	print ( sem )
	print ( "Sleep for 10 seconds" )
	time.sleep(10)
	sem.release()
	print ( "sem released." )

print ( "Acquire semaphore=user:4711" )
print ( "  asynchronous mode" )

sem = Semaphore ( "user:4711" )
sem.acquire ( on_acquired )
